bpfd/src/bpf.rs

```diff
@@ -159,23 +158,25 @@ impl<'a> BpfManager<'a> {
             .clone()
             .contains(BYTECODE_IMAGE_CONTENT_STORE)
         {
-            get_bytecode_from_image_store(program.data().path.clone())?
+            get_bytecode_from_image_store(program.data().path.clone()).await?
         } else {
-            std::fs::read(program.data().path.clone())
-                .map_err(|e| BpfdError::Error(format!("can't read bytecode file from disk {e}")))?
+            read(program.data().path.clone()).await?
         };
 
         let mut ext_loader = BpfLoader::new()
             .extension(&program.data().section_name)
             .map_pin_path(map_pin_path.clone())
             .load(&program_bytes)?;
 
-        ext_loader
-            .program_mut(&program.data().section_name)
-            .ok_or_else(|| {
-                let _ = fs::remove_dir_all(map_pin_path);
-                BpfdError::SectionNameNotValid(program.data().section_name.clone())
-            })?;
+        match ext_loader.program_mut(&program.data().section_name) {
```

Comment:
Why the move from `.ok_or_else` here?

Reply:
Because you can't use async functions in `ok_or_else`